3ds2sdk/src/main/kotlin/com/stripe/android/stripe3ds2/service/StripeThreeDs2ServiceImpl.kt

```diff
@@ -200,3 +217,18 @@ class StripeThreeDs2ServiceImpl @VisibleForTesting internal constructor(
         private const val STRIPE_SDK_REFERENCE_NUMBER = "3DS_LOA_SDK_STIN_020100_00142"
     }
 }
+
+class AnalyticsSingleton private constructor() {
```

Comment:
This won't work because it will not survive process death. I discussed with the team and we decided a java service provider will be the best approach. Can you remove this and replace with a `ServiceLoader`?

Reply:
Yes sorry, thought I posted a comment about the Singleton to ask how your team wanted to handle that. 👍🏼

Reply:
@tjclawson-stripe Rewritten to be `ServiceLoader` based. Let me know if this matches what you were thinking, it's been a minute since I did `ServiceLoader` based injection.